File: CS/MainForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using CustomFontEditControl;

namespace CustomFontEdit
{
    public partial class MainForm : Form
    {
        DataTable SomeTable = new DataTable();
        public MainForm()
        {
            InitializeComponent();
        }

        private void MainForm_Load(object sender, EventArgs e)
        {
            SomeTable.Columns.Add("SomeString", typeof(string));
            SomeTable.Columns.Add("SomeFont", typeof(Font));
            //SomeTable.Columns.Add("SomeFont", typeof(string));

            SomeTable.Rows.Add(new object[] { "1There is some text", null });
            SomeTable.Rows.Add(new object[] { "2There is some text", null });
            SomeTable.Rows.Add(new object[] { "3There is some text", null });
            SomeTable.Rows.Add(new object[] { "4There is some text", null });
            SomeTable.Rows.Add(new object[] { "5There is some text", null });
            SomeTable.Rows.Add(new object[] { "6There is some text", null });

            gridControlMain.DataSource = SomeTable;
            gridViewMain.Columns["SomeString"].ColumnEdit = repositoryItemTextEdit1;
            gridViewMain.Columns["SomeFont"].ColumnEdit = repositoryItemCustomFontEdit1;
        }

        private void gridViewMain_RowCellStyle(object sender, DevExpress.XtraGrid.Views.Grid.RowCellStyleEventArgs e)
        {
            if (e.Column == gridViewMain.Columns["SomeString"])
            {
                Font NeighbourCellFont = gridViewMain.GetRowCellValue(e.RowHandle, gridViewMain.Columns["SomeFont"]) as Font;
                if (NeighbourCellFont != null)
                {
                    e.Appearance.Font = NeighbourCellFont;   
                }                
            }
        }
    }
}

File: VB/MainForm.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace CustomFontEdit

    Public Partial Class MainForm
        Inherits Form

        Private SomeTable As DataTable = New DataTable()

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs)
            SomeTable.Columns.Add("SomeString", GetType(String))
            SomeTable.Columns.Add("SomeFont", GetType(Font))
            'SomeTable.Columns.Add("SomeFont", typeof(string));
            SomeTable.Rows.Add(New Object() {"1There is some text", Nothing})
            SomeTable.Rows.Add(New Object() {"2There is some text", Nothing})
            SomeTable.Rows.Add(New Object() {"3There is some text", Nothing})
            SomeTable.Rows.Add(New Object() {"4There is some text", Nothing})
            SomeTable.Rows.Add(New Object() {"5There is some text", Nothing})
            SomeTable.Rows.Add(New Object() {"6There is some text", Nothing})
            gridControlMain.DataSource = SomeTable
            gridViewMain.Columns("SomeString").ColumnEdit = repositoryItemTextEdit1
            gridViewMain.Columns("SomeFont").ColumnEdit = repositoryItemCustomFontEdit1
        End Sub

        Private Sub gridViewMain_RowCellStyle(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Grid.RowCellStyleEventArgs)
            If e.Column Is gridViewMain.Columns("SomeString") Then
                Dim NeighbourCellFont As Font = TryCast(gridViewMain.GetRowCellValue(e.RowHandle, gridViewMain.Columns("SomeFont")), Font)
                If NeighbourCellFont IsNot Nothing Then
                    e.Appearance.Font = NeighbourCellFont
                End If
            End If
        End Sub
    End Class
End Namespace
